From 7a5be2a042f3c1d55c5599ad819333a2150955ff Mon Sep 17 00:00:00 2001
From: Peter Korsgaard <peter@korsgaard.com>
Date: Fri, 4 Mar 2016 16:47:38 +0100
Subject: [PATCH 2/2] busybox: tweak IPv6/largefile handling

Fixes #8736

When IPv6 and largefile options were removed from Buildroot, the code to
force these options in busybox were still left in.

There's no strong reason to forcefully enable these options (only to disable
options if the system cannot support it like we do for nommu), so instead
enable the options in our default defconfig, allowing people to override
this if they use a custom config.

While we're at it, enable the prefer-ipv4 option so network applets like
ntpd doesn't fail when dual stacked hosts are resolved from a system without
IPv6 support enabled in the kernel.

Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
---
 package/busybox/busybox.mk | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/package/busybox/busybox.mk b/package/busybox/busybox.mk
index 7f6dbd6..7c904c8 100644
--- a/package/busybox/busybox.mk
+++ b/package/busybox/busybox.mk
@@ -107,16 +107,6 @@ define BUSYBOX_SET_MMU
 endef
 endif
 
-define BUSYBOX_SET_LARGEFILE
-	$(call KCONFIG_ENABLE_OPT,CONFIG_LFS,$(BUSYBOX_BUILD_CONFIG))
-	$(call KCONFIG_ENABLE_OPT,CONFIG_FDISK_SUPPORT_LARGE_DISKS,$(BUSYBOX_BUILD_CONFIG))
-endef
-
-define BUSYBOX_SET_IPV6
-	$(call KCONFIG_ENABLE_OPT,CONFIG_FEATURE_IPV6,$(BUSYBOX_BUILD_CONFIG))
-	$(call KCONFIG_ENABLE_OPT,CONFIG_FEATURE_IFUPDOWN_IPV6,$(BUSYBOX_BUILD_CONFIG))
-endef
-
 # If we're using static libs do the same for busybox
 ifeq ($(BR2_STATIC_LIBS),y)
 define BUSYBOX_PREFER_STATIC
@@ -219,8 +209,6 @@ endef
 
 define BUSYBOX_KCONFIG_FIXUP_CMDS
 	$(BUSYBOX_SET_MMU)
-	$(BUSYBOX_SET_LARGEFILE)
-	$(BUSYBOX_SET_IPV6)
 	$(BUSYBOX_PREFER_STATIC)
 	$(BUSYBOX_SET_MDEV)
 	$(BUSYBOX_SET_CRYPT_SHA)
-- 
1.9.1

